pkg/addon/common.go

```diff
@@ -141,9 +145,16 @@ func GetAndAddAgent(
 		return fmt.Errorf("failed getting the %v agent addon: %w", addonName, err)
 	}
 
-	agentAddon = &PolicyAgentAddon{agentAddon}
+	clusterClient, err := clusterv1client.NewForConfig(controllerContext.KubeConfig)
+	if err != nil {
+		return err
+	}
+
+	clusterInformers := clusterv1informers.NewSharedInformerFactory(clusterClient, 10*time.Minute)
```

Comment:
not sure if it is because we did not call clusterInformers.Start(ctx.Done()) , please check.

Reply:
This seems to have done it--thanks for the pointer!

```
go clusterInformers.Cluster().V1().ManagedClusters().Informer().Run(ctx.Done())
```